server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardLoopIntegrationTest.java

```diff
@@ -0,0 +1,247 @@
+        filteringManagementProbe.defineRulesForUser(ALICE,
+            Optional.empty(),
+            Rule.builder()
+                .id(Rule.Id.of("1"))
+                .name("rule 1")
+                .conditionGroup(Rule.ConditionGroup.of(Rule.ConditionCombiner.AND, Rule.Condition.of(FROM, NOT_CONTAINS, "AAA")))
+                .action(Rule.Action.builder().setAppendInMailboxes(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of()))
+                    .setWithKeywords(ImmutableList.of())
+                    .setForward(Optional.of(Rule.Action.Forward.of(ImmutableList.of(BOB.asMailAddress()), true)))
+                    .build())
+                .build());
+        filteringManagementProbe.defineRulesForUser(BOB,
+            Optional.empty(),
+            Rule.builder()
+                .id(Rule.Id.of("1"))
+                .name("rule 1")
+                .conditionGroup(Rule.ConditionGroup.of(Rule.ConditionCombiner.AND, Rule.Condition.of(FROM, NOT_CONTAINS, "AAA")))
+                .action(Rule.Action.builder().setAppendInMailboxes(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of()))
+                    .setWithKeywords(ImmutableList.of())
+                    .setForward(Optional.of(Rule.Action.Forward.of(ImmutableList.of(CEDRIC.asMailAddress()), false)))
+                    .build())
+                .build());
+        filteringManagementProbe.defineRulesForUser(CEDRIC,
+            Optional.empty(),
+            Rule.builder()
+                .id(Rule.Id.of("1"))
+                .name("rule 1")
+                .conditionGroup(Rule.ConditionGroup.of(Rule.ConditionCombiner.AND, Rule.Condition.of(FROM, NOT_CONTAINS, "AAA")))
+                .action(Rule.Action.builder().setAppendInMailboxes(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of()))
+                    .setWithKeywords(ImmutableList.of())
+                    .setForward(Optional.of(Rule.Action.Forward.of(ImmutableList.of(ALICE.asMailAddress()), false)))
+                    .build())
+                .build());
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER.asString(), PASSWORD)
+            .sendMessage(SENDER.asString(), ALICE.asString());
+
+        Awaitility.waitAtMost(1L, TimeUnit.HOURS).until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 2L);
+
+        SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+            List<Mail> mails = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
+                .filter(Throwing.predicate(mail -> mail.getRecipients().contains(ALICE.asMailAddress())))
+                .collect(ImmutableList.toImmutableList());
+
+            softly.assertThat(mails.get(0).getRecipients()).containsOnly(ALICE.asMailAddress());
+            softly.assertThat(mails.get(1).getRecipients()).containsOnly(ALICE.asMailAddress());
```

Comment:
When reading the ticket it seems Cedric should not send the mail back to alice (cause loop). One mail is because local copy, the second one I'm surprised though?

Reply:
Initially I could not figure out how to prevent Alice from getting a duplicate mail. Just fixed it. Check the newest commit.

Reply:
Cedric does not keep a copy

Reply:
cf https://github.com/apache/james-project/pull/1758#discussion_r1368334183